src/olympia/versions/tests/test_views.py

```diff
@@ -916,3 +926,11 @@ def test_download_for_admin(self):
         # Even deleted!
         self.addon.delete()
         assert self.client.get(self.url).status_code == 200
+
+
+class TestDownloadSourceSessionAPIAuth(TestDownloadSource, APILoginMixin):
```

Comment:
Subtle bug: `APILoginMixin` should be first, otherwise you're using the implementation for `login` provided by `TestDownloadSource`, which uses regular auth.

You can verify this by adding a `breakpoint()` inside `APILoginMixin.login()` and running tests for that class, it will never be triggered without changing that ordering.